test/plugin_helper/test_child_process.rb

```diff
@@ -232,30 +232,26 @@ def configure(conf)
   end
 
   test 'can execute external command at just once, which runs forever' do
-    m = Mutex.new
     ary = []
     Timeout.timeout(TEST_DEADLOCK_TIMEOUT) do
       ran = false
-      args = ["-e", "while sleep #{TEST_WAIT_INTERVAL_FOR_LOOP}; puts 1; STDOUT.flush; end"]
+      args = ["-e", "while sleep 0.2; puts 1; STDOUT.flush; end"]
       @d.child_process_execute(:t3, "ruby", arguments: args, mode: [:read]) do |io|
-        m.lock
         ran = true
         begin
           while @d.child_process_running? && line = io.readline
             ary << line
           end
         rescue
           # ignore
-        ensure
-          m.unlock
         end
       end
-      sleep TEST_WAIT_INTERVAL_FOR_BLOCK_RUNNING until m.locked? || ran
-      sleep TEST_WAIT_INTERVAL_FOR_LOOP * 10
+      sleep TEST_WAIT_INTERVAL_FOR_BLOCK_RUNNING until ran
+      sleep 1
+
       @d.stop # nothing occurs
       @d.shutdown
-
-      assert{ ary.size > 5 }
+      assert { ary.size >= 4 }
```

Comment:
This assertion is sometimes unstable on my environment.

```log
     251: 
     252:       @d.stop # nothing occurs
     253:       @d.shutdown
  => 254:       assert { ary.size >= 4 }
     255: 
     256:       @d.close
     257: 
~/GitHub/fluentd/test/plugin_helper/test_child_process.rb:254:in `block (2 levels) in <class:ChildProcessTest>'
Failure: test: can execute external command at just once, which runs forever(ChildProcessTest):
        assert { ary.size >= 4 }
                 |   |    |
                 |   |    false
                 |   2
                 ["1\n", "1\n"]
=========================================================================================================================================
```

Reply:
How about this? https://github.com/fluent/fluentd/pull/2770/commits/e256bb7932cf6c5a6c27f09c00a27c1d2d8abbe1

Reply:
Yeah, the testcase got stable.
But the other testcase gets unstable. Could you check it?
https://travis-ci.org/fluent/fluentd/jobs/639305592#L573

Reply:
I made a mistake... `ran` is never called in e256bb7... setting run true when it can read data for the first time. https://github.com/fluent/fluentd/pull/2770/commits/64c2c5a5d69ff8c72acfce0f79465b5937cc2d5a

Reply:
I've confirmed that these testcase instabilities are fixed.